### ISSUE TITLE
[Bug] PSG envelopes are too loud [sf#567]

### ISSUE THREAD

**Issue by openMSX-import** [openMSX/openMSX#958]
**Reported by grauw on 2015-02-22 18:05 UTC**
As reported by Xyz_39808:

“OpenMSX psg is really broken when it comes to the hardware envelope. It plays like 3 times louder than could be possible even if using it on every channel at once. At least, the turboR machine I emulate does.”

Compare at the 123 seconds mark, when the high frequency envelope modulation kicks in:

https://app.box.com/s/bjarqjt3a77i9fjpihvvwx4jq5evgfd8 (hardware)
https://dl.dropboxusercontent.com/u/39060235/openmsx.wav (openMSX)

On the real MSX the volume of the bass drum and envelope-driven sawtooth bass line stays nicely in balance, whereas on openMSX the bass drum really gets drowned out by the sawtooth.

You can test the song by running the attached disk image Xyz%20Disk.dsk.

Then after it boots, run "mgsdrv\mgsdrv.com /z", and after that "%mgsel%" to start the replayer. The test song is WELLBE~1.MGS in the root of the folder, play it. Hold F3 to fast forward. The P counter is the timer. Channels can be muted by pressing 1, 2 and 3.

Xyz_39808 further writes: “It happens for both low and high frequency ranges. When enabling the hardware envelope, it ramps the volume in a range that is outside what could normally be done on the SSG. It exceeds the normal volume.”

**Comment by openMSX-import** [comment]
**Updated by grauw on 2015-02-22 18:05 UTC**
- **summary**: PSG  --> PSG envelopes are too loud

**Comment by openMSX-import** [comment]
**Commented by grauw on 2015-02-22 18:05 UTC**
Forgot to write a proper title

**Comment by oh-ren** [comment]
Finally got around uploading those recordings: https://oh-ren.github.io/msx/openMSX/issue-958/

The *real HB-F1XD.wav* snippet is simply cut from the wav provided by Xyz_39808 above.

I'm not sure how to explain/interpret those differences in waveform (screen dump from Reaper)?

Since this is an imported issue, @grauw won't be notified (but now he is.. ;))

**Comment by oh-ren** [comment]
@openMSX (does this work? :P) can you confirm you noticed my update?

So my observation/conclusion: the MAME PSG emulation doesn't seem to suffer (as much?) from the bug reported here. It does sound more muffled than openMSX.
blueMSX suffers from the same issue (and some percussion/hi-hats sound different on it), and fMSX showed similar + other issues I believe (+ sounding rather poor.)

**Comment by MBilderbeek** [comment]
Yes I read you. Thanks for those recordings! Isn't the problem more that on the real MSX the bass is a LOT louder? Also visible in these visuals. Could this be due to the analog filtering that is behind the real MSX in the audio circuit?

**Comment by oh-ren** [comment]
> Isn't the problem more that on the real MSX the bass is a LOT louder?

I don't think so. If you level the volumes, so all recordings are more or less the same loudness to your ears (at say the 20s mark), you clearly hear a huge difference at the 29s mark.

To my ears in openMSX the kick drum stays the same level, only the sawtooth bass line (kicking in @ 29s) is really, really taking a huge step in loudness.

The MAME recording seems closer the the real hw recording in this regard, but still the sawtooth seems to be too loud here OR (/and) the bass kick is somehow softer in the rendering at that point (to my ears it does seem that way) .

In the real recording you can clearly distinguish the two. Bass kick remaining at the same loudness level, and the sawtooth seems balanced with the rest volume-wise.

Perhaps @grauw has some observations in this regard as well (and perhaps observed some (more or less) similar cases whilst working with openMSX)?

I guess it might help if someone else actually listens to it on his/her real MSX.. (and pref. record it of course, or otherwise just report back the findings.) The disk with the song is still available as attachment to [the original SF issue](https://sourceforge.net/p/openmsx/bugs/567/).
